extensions/basic_auth/config/gateway-filter.yaml

```diff
@@ -0,0 +1,56 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+  name: basic-auth-filter
+  namespace: istio-system
+spec:
+  configPatches:
+  - applyTo: HTTP_FILTER
+    match:
+      context: GATEWAY
+      listener:
+        filterChain:
+          filter:
+            name: envoy.http_connection_manager
+      proxy:
+        proxyVersion: ^1\.8.*
+    patch:
+      operation: INSERT_BEFORE
+      value:
+        name: example-filter
+        typed_config:
+          '@type': type.googleapis.com/udpa.type.v1.TypedStruct
+          type_url: type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm
+          value:
+            config:
+              configuration:
+                '@type': type.googleapis.com/google.protobuf.StringValue
+                value: |
+                  {
+                    "basic_auth_rules": [
+                      {
+                        "prefix": "/",
+                        "request_methods": ["GET", "POST"],
+                        "credentials": [
+                          "ok:test",
+                          "admin:admin",
+                          "admin2:admin2"
+                        ]
+                      }
+                    ]
+                  }
+              vm_config:
+                vm_id: basic_auth
+                code:
+                  remote:
+                    http_uri:
+                      uri: https://storage.googleapis.com/istio-ecosystem/wasm-extensions/basic-auth/1.8.0.wasm
```

Comment:
Note since we've not cut a new release, so we 1.8.0.wasm has not yet really been pushed. If you want to give it a try, please use the following URL:
```yaml
  remote:
    http_uri:
      uri: https://storage.googleapis.com/istio-ecosystem/wasm-extensions/basic-auth/0cca8f60b157d2a6ea0eaf114717992e6a6afea3.wasm
      cluster: google-storage
      timeout: 10s
    sha256: b771dd1a96869a03f2a2cf812ff4f37f75939ca9ac9b432e89390caf33f581c4
```